## Price Change
Ethereum has been undergoing a period of big price fluctuations in 2022. We will try to understand how it happened by checking the percentage change of the price of Ether in dollars in each day, and getting an average change in the whole month, and comparing with the percentage change from one month to another. Data from the daily price of Ether was collected from https://etherscan.io/chart/etherprice

This approach is based on what Aspembitova AT, Feng L, Chew LY
 did in "Behavioral structure of users in cryptocurrency market"

In [9]:
#Imports only
import pandas as pd
import datetime

In [19]:
df = pd.read_csv('CSVs/export-EtherPrice05092022.csv')
df['Date(UTC)'] = pd.to_datetime(df['Date(UTC)'], format = '%m/%d/%Y') #Date has to be in datetime object to be manipulated
df = df[df['Date(UTC)'].dt.date >= datetime.date(2022,1,1)] #Filtering only results from 2022
df['Month'] = [i.month for i in df['Date(UTC)']] #Month comparasions will be important
display(df)

Date(UTC)  UnixTimeStamp    Value  Month
2347 2022-01-01     1640995200  3766.74      1
2348 2022-01-02     1641081600  3830.92      1
2349 2022-01-03     1641168000  3765.16      1
2350 2022-01-04     1641254400  3785.67      1
2351 2022-01-05     1641340800  3536.77      1
...         ...            ...      ...    ...
2589 2022-08-31     1661904000  1554.16      8
2590 2022-09-01     1661990400  1585.98      9
2591 2022-09-02     1662076800  1574.02      9
2592 2022-09-03     1662163200  1557.54      9
2593 2022-09-04     1662249600  1579.06      9

[247 rows x 4 columns]

In [23]:
#Now, let's create a new df to check monthly changes as a whole
df_monthly = df[['Month','Value']].groupby(['Month']).sum()
display(df_monthly.pct_change()*100)

Value
Month           
1            NaN
2     -15.155138
3      11.037062
4       4.133269
5     -27.038383
6     -38.837387
7       1.634585
8      25.048795
9     -88.045412

We can see some results here. January doesn't have a previous month to compare with, and September doesn't have a just comparasion yet, as it is a ongoing month as of the date of the collected dataset.

|  Months   |  % Total  | Trend  |
|:---------:|:----------:|:----------:|
|  January  | Don't know | Don't know |
| February  |  -15.15%   |  Decrease  |
|   March   |   11.03%   |  Increase  |
|   April   |   4.13%    |   Stable   |
|    May    |  -27.04%   |  Decrease  |
|   June    |  -38.84%   |  Decrease  |
|   July    |   1.63%    |   Stable   |
|  August   |   25.05%   |  Increase  |
| September | Don't know | Don't know |

In [30]:
#Now for the daily analysis
df_avg_month = df[['Value']].pct_change()*100
df_avg_month['Month'] = df['Month'] #Retrieving the month to group it again
df_avg_month = df_avg_month.groupby(['Month']).mean()
display(df_avg_month)

Value
Month          
1     -1.026895
2      0.405443
3      0.428405
4     -0.568190
5     -0.955328
6     -1.787745
7      1.620728
8     -0.142924
9      0.406991

Now, we can add more information to the previous table.

|  Months   |  % Total   | % Average  |   Trend    |
|:---------:|:----------:|:----------:|:----------:|
|  January  | Don't know |   -1.03%   | Don't know |
| February  |  -15.15%   |   0.41%    |  Decrease  |
|   March   |   11.03%   |   0.43%    |  Increase  |
|   April   |   4.13%    |   -0.57%   |   Stable   |
|    May    |  -27.04%   |   -0.96%   |  Decrease  |
|   June    |  -38.84%   |   -1.79%   |  Decrease  |
|   July    |   1.63%    |   1.62%    |   Stable   |
|  August   |   25.05%   |   -0.43%   |  Increase  |
| September | Don't know |   0.41%    | Don't know |

Note that we **can** say something about January and September now, since we are just comparing an averaged information within the same month.

Based on that information about the months, the period of interesting analysis would probably be from April/May (the moment when the value starts to drop a lot) to August/September (the moment when it starts to go back up and approaching The Merge).